frappe/desk/like.py

```diff
@@ -31,7 +31,15 @@ def _toggle_like(doctype, name, add, user=None):
 		user = frappe.session.user
 
 	try:
-		liked_by, owner = frappe.db.get_value(doctype, name, ["_liked_by", "owner"])
+		if is_single(doctype):
+			values = frappe.db.get_singles_dict(doctype)
+			if values.has_key("_liked_by"):
+				liked_by = frappe.db.get_value(doctype, None, "_liked_by")
+			else:
+				frappe.db.sql("insert into tabSingles(doctype, field, value) values (%s, %s, Null)", (doctype, "_liked_by"))
```

Comment:
Inserting a new row is not necessary. If the row is not present, it should be assumed to be `None`